Kickstarter-iOS/Views/Controllers/CommentsViewControllerTests.swift

```diff
@@ -16,10 +16,47 @@ internal final class CommentsViewControllerTests: TestCase {
     super.tearDown()
   }
 
-  func testView() {
+  func testView_CurrentUser_LoggedOut() {
     Language.allLanguages.forEach { language in
-      withEnvironment(language: language) {
-        let controller = CommentsViewController()
+      withEnvironment(currentUser: nil, language: language) {
+        let controller = CommentsViewController.configuredWith(project: .template)
+        let (parent, _) = traitControllers(device: .phone4_7inch, orientation: .portrait, child: controller)
+        parent.view.frame.size.height = 1_100
+
+        self.scheduler.run()
+
+        controller.tableView.layoutIfNeeded()
+        controller.tableView.reloadData()
+
+        FBSnapshotVerifyView(parent.view, identifier: "Comments - lang_\(language)")
+      }
+    }
+  }
+
+  func testView_CurrentUser_LoggedIn_IsBacking_True() {
+    let project = Project.template
+      |> \.personalization.isBacking .~ true
+
+    Language.allLanguages.forEach { language in
+      withEnvironment(currentUser: .template, language: language) {
+        let controller = CommentsViewController.configuredWith(project: project)
+        let (parent, _) = traitControllers(device: .phone4_7inch, orientation: .portrait, child: controller)
+        parent.view.frame.size.height = 1_100
+
+        self.scheduler.run()
+
+        controller.tableView.layoutIfNeeded()
+        controller.tableView.reloadData()
+
+        FBSnapshotVerifyView(parent.view, identifier: "Comments - lang_\(language)")
+      }
+    }
+  }
+
+  func testView_CurrentUser_LoggedIn_IsBacking_False() {
```

Comment:
great job on tests!

Reply:
Always nice to get the coverage right on the first go hehe